lib/ansible/galaxy/collection.py

```diff
@@ -584,6 +591,82 @@ def build_collection(collection_path, output_path, force):
     return collection_output
 
 
+def _collection_matches_requirement(collection, requirement, parent=None):
+    try:
+        collection.add_requirement(parent, requirement)
+    except AnsibleError:
+        return False
+    else:
+        return True
+
+
+def find_matching_collections(namespace, name, requirement, search_paths, first_found=False, fallback_metadata=False, parent=None, require_match=False):
+    found = non_artifact_found = False
+    for search_path in search_paths:
+        display.vvvv(
+            "Looking for '%s.%s:%s' in the collection search path %s" % (namespace, name, requirement, to_text(search_path, errors='surrogate_or_strict'))
+        )
+
+        b_collection_path = to_bytes(os.path.join(search_path, namespace, name), errors='surrogate_or_strict')
+
+        if not os.path.isdir(b_collection_path):
+            continue
+
+        if not os.path.isfile(os.path.join(b_collection_path, b'MANIFEST.json')) and not fallback_metadata:
+            non_artifact_found = True
+            continue
+
+        local_collection = CollectionRequirement.from_path(b_collection_path, False, fallback_metadata=fallback_metadata, parent=parent)
+
+        if not _collection_matches_requirement(local_collection, requirement, parent):
+            continue
+
+        found = True
+        display.vvvv("Found collection '%s.%s:%s'" % (namespace, name, requirement))
+        yield local_collection
+        if first_found:
+            break
+
+    if require_match and not found:
+        if not fallback_metadata and non_artifact_found:
+            raise AnsibleError(
+                message="Collection %s.%s does not appear to have a MANIFEST.json. " % (namespace, name) +
+                "A MANIFEST.json is expected if the collection has been built and installed via ansible-galaxy.")
+        else:
+            raise AnsibleError(message='Collection %s.%s is not installed in any of the collection paths.' % (namespace, name))
+    elif not found:
+        display.display('%s.%s:%s is not installed' % (namespace, name, requirement))
+
+
+def remove_collections(collections, search_paths, ignore_errors, no_deps, parent=None):
+    with _display_progress():
+        for collection in collections:
+            b_collection = to_bytes(collection[0], errors='surrogate_or_strict')
+            if os.path.isfile(b_collection) or urlparse(collection[0]).scheme.lower() in ['http', 'https'] or len(collection[0].split('.')) != 2:
+                raise AnsibleError(message="'%s' is not a valid collection name. The format namespace.name is expected." % collection[0])
+
+            collection_name = collection[0]
+            namespace, name = collection_name.split('.')
+            collection_version = collection[1]
+
+            # do we want to remove all or first found? should it be a toggle?
```

Comment:
One suggestion is that we only remove automatically if 1 was found but prompt the user if multiple were found. It would be something like

```
Found the following collections that match 'community.general'
    1 - /home/user/.ansible/collections/ansible_collections/community/general
    2 - /etc/ansible/collections/ansible_collections/community/general
    3 - /tmp/ansible_collections/community/general

Type the comma separate list of numbers to remove (a for all):
```

We could add an `--all` argument as well if we wanted a non-interactive run but we do run the risk of having an arg that most users want set by default.

Reply:
I implemented this to err on the side of caution as discussed. I used @webknjaz's suggestion of using the `--force` flag for non-interactive runs.